toolchain/check/testdata/class/fail_modifiers.carbon

```diff
@@ -8,6 +8,10 @@
 // TIP: To dump output, run:
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/class/fail_modifiers.carbon
 
+// CHECK:STDERR: fail_modifiers.carbon:[[@LINE+11]]:9: error: `abstract` not allowed on `class` declaration
+// CHECK:STDERR: private abstract private class DuplicatePrivate;
```

Comment:
Feels better to have separate tests for "abstract not allowed on declaration" from "modifier repeated" and so on. Perhaps make these tests use a class definition (`{}` instead of `;`), now that those are different? (similarly other tests)